Source/DafnyCore/Dafny.atg

```diff
@@ -96,7 +96,11 @@ void CheckDeclModifiers(ref DeclModifierData dmod, string declCaption, AllowedDe
   }
   if (dmod.IsGhost) {
     if ((allowed & AllowedDeclModifiers.AlreadyGhost) != 0) {
-      SemErr(dmod.GhostToken, $"{declCaption} cannot be declared 'ghost' (it is 'ghost' by default)");
+      if (declCaption.Contains("-by-method")) {
+        SemErr(dmod.GhostToken, $"{declCaption} has a ghost function body and a non-ghost method body; declaring it 'ghost' is incorrect, confusing and not permitted");
```

Comment:
After the `;`, I suggest simply saying

    a {declCaption} declaration does not use the 'ghost' keyword

Reply:
Fixed